### PR TITLE
kops job configs: Refactor to remove common ci-kubernetes-e2e-kops-aws.env

### PR DESCRIPTION
**Pull request by zmerlynn**
This PR does two things:

* There were a lot of jobs sharing ci-kubernetes-e2e-kops-aws.env, but
it has a dangerous setting that changes latest-ci-green.txt unless
overridden. Move the common settings to flags and kops_aws.env and
un-share.

* Switch to local mode for all remaining docker kops jobs.

### CODE REVIEW

File: jobs/config.json [hunk 1]
```diff
@@ -8157,9 +8157,9 @@
       "--env-file=jobs/platform/kops_aws.env",
       "--env-file=jobs/env/ci-kubernetes-e2e-kops-aws.env",
       "--extract=ci/latest",
-      "--mode=docker",
```
Comment (krzyzacy) [inline jobs/config.json]: you'll need to move the config from https://github.com/kubernetes/test-infra/blob/master/jenkins/job-configs/kubernetes-jenkins/bootstrap-ci.yaml#L58-L71 to around https://github.com/kubernetes/test-infra/blob/master/prow/config.yaml#L15079

File: jobs/config.json [hunk 2]
```diff
@@ -10596,7 +10592,7 @@
       "--ginkgo-parallel",
```
Comment (krzyzacy) [inline jobs/config.json]: (two lines above, remove the `pull-kubernetes-e2e-kops-aws-prow.env` reference?)

File: jobs/config.json (hunk 1 above)
Comment (zmerlynn) [reply jobs/config.json]: Oh, right. Der.

**Comment by krzyzacy** [comment]
/lgtm
(hopefully)

**Comment by spiffxp** [comment]
https://k8s-testgrid.appspot.com/kubernetes-presubmits#pull-kubernetes-e2e-kops-aws

I think this broke the kops presubmit

**Comment by zmerlynn** [comment]
I'm on mobile but feel free to just revert, can do after the release.

File: jobs/config.json [hunk 3]
```diff
@@ -10575,7 +10574,7 @@
       "--mode=docker",
       "--provider=aws",
       "--stage=gs://kubernetes-release-pull/ci/pull-kubernetes-e2e-kops-aws",
-      "--test_args=--ginkgo.skip=\\[Slow\\]|\\[Serial\\]|\\[Disruptive\\]|\\[Flaky\\]|\\[Feature:.+\\]|\\[HPA\\]|Dashboard|Services.*functioning.*NodePort",
+      "--test_args=--ginkgo.flakeAttempts=2 --ginkgo.skip=\\[Slow\\]|\\[Serial\\]|\\[Disruptive\\]|\\[Flaky\\]|\\[Feature:.+\\]|\\[HPA\\]|Dashboard|Services.*functioning.*NodePort",
```
Comment (krzyzacy) [inline jobs/config.json]: whoops - add `--ginkgo-parallel` here

**Pull request merged**